src/webgpu/api/validation/createRenderPipeline.spec.ts

```diff
@@ -101,97 +101,122 @@ class F extends ValidationTest {
+      if (_success) {
+        this.shouldResolve(this.device.createRenderPipelineAsync(descriptor).then(callback));
+      } else {
+        // If creating render pipeline is expected to fail, we just skip the callback
+        this.shouldReject('OperationError', this.device.createRenderPipelineAsync(descriptor));
+      }
+    } else {
+      if (_success) {
+        const pipeline = this.device.createRenderPipeline(descriptor);
+        callback(pipeline);
+      } else {
+        this.expectValidationError(() => {
+          this.device.createRenderPipeline(descriptor);
+        });
+      }
+    }
+  }
 }
 
 export const g = makeTestGroup(F);
 
-g.test('basic_use_of_createRenderPipeline').fn(t => {
-  const descriptor = t.getDescriptor();
-
-  t.device.createRenderPipeline(descriptor);
-});
+g.test('basic_use_of_createRenderPipeline')
+  .params(u => u.combine('isAsync', [false, true]))
+  .fn(async t => {
+    const { isAsync } = t.params;
+    const descriptor = t.getDescriptor();
 
-g.test('at_least_one_color_state_is_required').fn(async t => {
-  const goodDescriptor = t.getDescriptor({
-    targets: [{ format: 'rgba8unorm' }],
+    t.doCreateRenderPipelineTest(isAsync, true, descriptor);
   });
 
-  // Control case
-  t.device.createRenderPipeline(goodDescriptor);
+g.test('at_least_one_color_state_is_required')
+  .params(u => u.combine('isAsync', [false, true]))
+  .fn(async t => {
+    const { isAsync } = t.params;
 
-  // Fail because lack of color states
-  const badDescriptor = t.getDescriptor({
-    targets: [],
-  });
+    const goodDescriptor = t.getDescriptor({
+      targets: [{ format: 'rgba8unorm' }],
+    });
+
+    // Control case
+    t.doCreateRenderPipelineTest(isAsync, true, goodDescriptor);
 
-  t.expectValidationError(() => {
-    t.device.createRenderPipeline(badDescriptor);
+    // Fail because lack of color states
+    const badDescriptor = t.getDescriptor({
+      targets: [],
+    });
+
+    t.doCreateRenderPipelineTest(isAsync, false, badDescriptor);
   });
-});
 
 g.test('color_formats_must_be_renderable')
-  .params(u => u.combine('format', kTextureFormats))
+  .params(u =>
+    u.combine('isAsync', [false, true]).combine('format', kTextureFormats as GPUTextureFormat[])
+  )
   .fn(async t => {
-    const format: GPUTextureFormat = t.params.format;
+    const { isAsync, format } = t.params;
     const info = kTextureFormatInfo[format];
     await t.selectDeviceOrSkipTestCase(info.feature);
 
     const descriptor = t.getDescriptor({ targets: [{ format }] });
 
     if (info.renderable && info.color) {
       // Succeeds when color format is renderable
-      t.device.createRenderPipeline(descriptor);
+      t.doCreateRenderPipelineTest(isAsync, true, descriptor);
     } else {
       // Fails because when format is non-renderable
-      t.expectValidationError(() => {
-        t.device.createRenderPipeline(descriptor);
-      });
+      t.doCreateRenderPipelineTest(isAsync, false, descriptor);
     }
```

Comment:
nit: if and else branch are almost the same now, instead do something like `const success = info.renderable && info.color; t.doCreateRenderPipelineTest(isAsync, success, descriptor);`